dataplane/base.h

```diff
@@ -14,13 +14,48 @@
 namespace dataplane::base
 {
 
+class PortMapper
+{
+	static constexpr tPortId INVALID_PORT_ID = std::numeric_limits<tPortId>::max();
+	uint16_t ports_count_ = 0;
+	tPortId dpdk_ports_[CONFIG_YADECAP_PORTS_SIZE] = {INVALID_PORT_ID}; // logical to dpdk
+	tPortId logical_ports_[std::numeric_limits<tPortId>::max() + 1] = {INVALID_PORT_ID}; // dpdk to logical
+public:
+	uint16_t size() const { return ports_count_; }
+	std::optional<tPortId> Register(tPortId dpdk_port)
+	{
+		if (ports_count_ < CONFIG_YADECAP_PORTS_SIZE)
+		{
+			if (logical_ports_[dpdk_port] == INVALID_PORT_ID)
+			{
+				logical_ports_[dpdk_port] = ports_count_;
+				dpdk_ports_[ports_count_] = dpdk_port;
+				return std::optional<tPortId>{ports_count_++};
+			}
+			else
+			{
+				YANET_LOG_ERROR("Duplicate dpdk port id provided to PortMapper");
+				return {};
+			}
+		}
+		else
+		{
+			YANET_LOG_ERROR("CONFIG_YADECAP_PORTS_SIZE exceeded");
+			return {};
+		}
+	}
+	tPortId ToDpdk(tPortId logical) const { return logical < CONFIG_YADECAP_PORTS_SIZE ? dpdk_ports_[logical] : INVALID_PORT_ID; }
+	tPortId ToLogical(tPortId dpdk) const { return logical_ports_[dpdk]; }
+	bool ValidDpdk(tPortId dpdk) const { return logical_ports_[dpdk] != INVALID_PORT_ID; }
+	bool ValidLogical(tPortId logical) const { return logical < INVALID_PORT_ID; }
```

Comment:
What will be the meaning of `[[nodiscard]]` here? Do we require an action from the caller? Do we return an error that needs to be handled? Do we return a unique handle to something that is lost forever if the caller discards it?

Reply:
No, I'm just used to putting `[[nodiscard]]` on such small functions that return a certain result.
But I agree that it may be redundant

Reply:
IMHO, blanketly using `[[nodiscard]]` detracts from the message that this functions(specifically) return value should not be discarded. In general any functions return value should not be discarded, but this functions -- especially.

There seems to be no consensus on the matter even in C++20 STL: for `vector` `empty()` is `[[nodiscard]]` while `size()` is not.

Reply:
I think we can resolve this conversation, as it is a topic for debate. Microsoft, for example, marked _a lot_ of their functions in the STL as [[nodiscard]]: https://github.com/microsoft/STL/issues/206

When we adopt a specific coding standard, we could specify what actions to take regarding [[nodiscard]]. A linter like clang-tidy could easily mark them if we agree on adding them, or remove them if we decide otherwise.

I'm not able to resolve the conversation myself; I guess I don't have the required permissions yet. :)